src/Servers/Kestrel/Core/src/Internal/Infrastructure/StringUtilities.cs

```diff
@@ -109,6 +113,179 @@ public static unsafe bool TryGetAsciiString(byte* input, char* output, int count
+                                Unsafe.Add(ref str, offset + 1) != (char)Unsafe.Add(ref bytes, offset + 1) ||
+                                Unsafe.Add(ref str, offset + 2) != (char)Unsafe.Add(ref bytes, offset + 2) ||
+                                Unsafe.Add(ref str, offset + 3) != (char)Unsafe.Add(ref bytes, offset + 3))
+                            {
+                                goto NotEqual;
+                            }
+
+                            offset += sizeof(int);
+                        }
+                    }
+                    else
+                    {
+                        // 32-bit: Loop ints by default
+                        while ((byte*)(offset + sizeof(int)) <= (byte*)count)
+                        {
+                            // Compare via simple widen from byte to char; note this will only be equal for ascii.
+                            if (Unsafe.Add(ref str, offset) != (char)Unsafe.Add(ref bytes, offset) ||
+                                Unsafe.Add(ref str, offset + 1) != (char)Unsafe.Add(ref bytes, offset + 1) ||
+                                Unsafe.Add(ref str, offset + 2) != (char)Unsafe.Add(ref bytes, offset + 2) ||
+                                Unsafe.Add(ref str, offset + 3) != (char)Unsafe.Add(ref bytes, offset + 3))
+                            {
+                                goto NotEqual;
+                            }
+
+                            offset += sizeof(int);
+                        }
+                    }
+                    if ((byte*)(offset + sizeof(short)) <= (byte*)count)
+                    {
+                        // Compare via simple widen from byte to char; note this will only be equal for ascii.
+                        if (Unsafe.Add(ref str, offset) != (char)Unsafe.Add(ref bytes, offset) ||
+                            Unsafe.Add(ref str, offset + 1) != (char)Unsafe.Add(ref bytes, offset + 1))
+                        {
+                            goto NotEqual;
+                        }
+
+                        offset += sizeof(short);
+                    }
+                    if ((byte*)offset < (byte*)count)
+                    {
+                        // Compare via simple widen from byte to char; note this will only be equal for ascii.
+                        if (Unsafe.Add(ref str, offset) != (char)Unsafe.Add(ref bytes, offset))
+                        {
+                            goto NotEqual;
+                        }
+                    }
+
+                    // End of input reached, there are no inequalities via widening; so the input bytes are both ascii
+                    // and a match to the string if it was converted via Encoding.ASCII.GetString(...)
+                    return true;
+                }
+
+                // Create a comparision vector for all bits being equal
+                var AllTrue = new Vector<ushort>(ushort.MaxValue);
```

Comment:
Does seem to be 1 i.e. returning 1,1,1, ... rather than 0xffff, 0xffff, 0xffff, ...

![image](https://user-images.githubusercontent.com/1142958/55369385-45b5c200-54ee-11e9-9e94-4267dff71812.png)

Reply:
Whereas MaxValue is
![image](https://user-images.githubusercontent.com/1142958/55369456-84e41300-54ee-11e9-9005-773d90214459.png)

Reply:
Left as `new Vector<ushort>(ushort.MaxValue)` as need all the bits set